### ISSUE TITLE
Loading Assets after the Companion is connected.

### ISSUE THREAD

**Issue by thequixotic** [mit-cml/appinventor-sources#503]
Currently in AI, after you connect the companion. Then you upload an image (asset) to the project (AI Client) the image won't be shown in the phone.
Users would have to reconnect the companion to see the asset on the phone. 

Because the image isn't transferred to the phone at all.
I stumbled this bug when importing Extensions after the Companion was connected.The extension wasn't being loaded into the phone. The reason being..
The assets are transferred only once (during the beginning of the connection)

I do have a fix for this. But I am not sure if this is meant to be fixed.
Is this is some security feature or something ? @jisqyv

**Comment by halatmit** [comment]
I don't think this was a security feature -- at least not an intentional
one.
I think you should fix it.

**Comment by jisqyv** [comment]
@thequixotic Nope, its just a bug!

**Comment by thequixotic** [comment]
Sorry @halatmit , @jisqyv  .

It was a false alarm. 
The assets are being transferred. There is an error though.

scheme.eval is only triggered when there is a change in the Yail code.

Sometimes scheme.eval is executed before the asset reaches the phone. 
When this happens, the asset is not seen on the phone. (because the asset wasn't really there when scheme.eval was executed)
So to see the asset on the phone, you will have to trigger another scheme.eval (by changing anything in the YA app)
And you can see the new asset.

I do have a fix for this (to hold the scheme eval code till the assets are reached on the phone) but it doesn't seem so necessary once the companion is properly connected.
You just have to make some change again to trigger the scheme.eval and see the new asset.

Sorry again, it was another issue. 
The solution to the problem must be similar to the one I made for cdk
(https://github.com/thequixotic/appinventor-cdk/commit/4b1bdd373b9ec22ab6d8bf92b023f6030540945d)

**Comment by halatmit** [comment]
Sounds like there should be a command to trigger scheme.eval

Jef, doe that make sense?



== Hal Abelson
hal@mit.edu
Prof. of Comp. Sci. and Eng.
MIT Dept. of Elec. Eng. and Comp. Sci.
MIT Media Lab, Second Class Faculty
